src/replication.c

```diff
@@ -2193,6 +2217,10 @@ void readSyncBulkPayload(connection *conn) {
             bg_unlink(server.rdb_filename);
         }
 
+        if (reclaimFilePageCache(server.repl_transfer_fd, 0, 0) == -1) {
```

Comment:
in theory we can maybe also incrementally release when parsing the rdb, but that's probably more complicated and we better skip that.